src/useHookFormMask.ts

```diff
@@ -12,7 +12,9 @@ export function useHookFormMask<
   return (fieldName: Path<T>, mask: Mask, options?: Options & D) => {
     if (!registerFn) throw new Error('registerFn is required');
 
-    const { ref, ...restRegister } = registerFn(fieldName);
+    const { setValueAs } = options || {};
+
+    const { ref, ...restRegister } = registerFn(fieldName, { setValueAs });
```

Comment:
We can pass all the props to register, no problem.
```suggestion
    const { ref, ...restRegister } = registerFn(fieldName, options);
```

Reply:
Passing all props to Hook Form they works like a charm, ignoring the inputmask props.

Reply:
I didn't thought of passing everything to register as well :D I just pushed the requested changes